src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -797,44 +800,82 @@ private function registerValidationConfiguration(array $config, ContainerBuilder
         }
     }
 
-    private function getValidatorMappingFiles(ContainerBuilder $container)
+    /**
+     * Get all validation files.
+     *
+     * @param ContainerBuilder $container A ContainerBuilder instance
+     * @param array            $files     A array reference of validation files
+     */
+    private function getValidatorMappingFiles(ContainerBuilder $container, array &$files)
     {
-        $files = array(array(), array());
-
         if (interface_exists('Symfony\Component\Form\FormInterface')) {
             $reflClass = new \ReflectionClass('Symfony\Component\Form\FormInterface');
-            $files[0][] = dirname($reflClass->getFileName()).'/Resources/config/validation.xml';
-            $container->addResource(new FileResource($files[0][0]));
+            $files['xml'][] = $file = dirname($reflClass->getFileName()).'/Resources/config/validation.xml';
+            $container->addResource(new FileResource($file));
         }
 
         $bundles = $container->getParameter('kernel.bundles');
         foreach ($bundles as $bundle) {
             $reflection = new \ReflectionClass($bundle);
             $dirname = dirname($reflection->getFileName());
 
-            if (is_file($file = $dirname.'/Resources/config/validation.xml')) {
-                $files[0][] = realpath($file);
+            if (is_file($file = $dirname.'/Resources/config/validation.yml')) {
+                $files['yml'][] = realpath($file);
                 $container->addResource(new FileResource($file));
             }
 
-            if (is_file($file = $dirname.'/Resources/config/validation.yml')) {
-                $files[1][] = realpath($file);
+            if (is_file($file = $dirname.'/Resources/config/validation.xml')) {
+                $files['xml'][] = realpath($file);
                 $container->addResource(new FileResource($file));
             }
 
             if (is_dir($dir = $dirname.'/Resources/config/validation')) {
-                foreach (Finder::create()->files()->in($dir)->name('*.xml') as $file) {
-                    $files[0][] = $file->getRealpath();
-                }
-                foreach (Finder::create()->files()->in($dir)->name('*.yml') as $file) {
-                    $files[1][] = $file->getRealpath();
-                }
-
+                $this->getValidatorMappingFilesFromDir($dir, $files);
                 $container->addResource(new DirectoryResource($dir));
             }
         }
+    }
 
-        return $files;
+    /**
+     * Search for validation files in directory.
+     *
+     * @param string $dir   The directory that contains the validation files
+     * @param array  $files A array reference of validation files
+     */
+    private function getValidatorMappingFilesFromDir($dir, array &$files)
+    {
+        foreach (Finder::create()->files()->in($dir)->name('/\.xml|yml$/') as $file) {
```

Comment:
Shouldn't it be `'/\.(xml|yml)$/'` ? 
(so `'/\.(xml|yml|yaml)$/'` with the `.yaml` extension)